.github/workflows/nextflow-tests.yml

```diff
@@ -73,11 +169,257 @@ jobs:
 
   summary:
     runs-on: ubuntu-latest
-    needs: run
-    if: ${{ !cancelled() }}
+    needs: [run, discover]
+    if: ${{ !cancelled() && needs.discover.result == 'success' && needs.discover.outputs.abort_early == 'false' }}
 
     steps:
       - uses: actions/github-script@v7
-        if: ${{ needs.run.result != 'success' && needs.run.result != 'skipped' }}
+        id: status
+        name: Determine required steps
         with:
-          script: core.setFailed('Tests failed!')
+          script: |
+            var needs = ${{ toJSON(needs) }}
+            var tests_failed = needs.run.result != 'success' &&
+              needs.run.result != 'skipped'
+
+            var should_comment = tests_failed &&
```

Comment:
I suggest sticking to just `var` or `let` if it's appropriate for this code (in addition to the use of `const` where appropriate).

Reply:
Done with ddf79e9.